src/state/slices/postsSlice.ts

```diff
@@ -14,31 +16,98 @@ export type Post = {
   parentPostId?: number;
 };
 
-const initialState: Post[] = [];
+type PostSliceType = {
+  posts: Post[];
+  expandedPost: Post;
+};
+
+const initialState: PostSliceType = {
+  posts: [],
+  expandedPost: {
+    displayName: "",
+    isLikedByCurrentUser: false,
+    numberOfLikes: 0,
+    numberOfReplies: 0,
+    numberOfReposts: 0,
+    postId: 0,
+    textContent: "",
+    timestamp: "",
+    username: "",
+    isRepost: false,
+    isQuotePost: false,
+  },
+};
 
 export const postsSlice = createSlice({
   name: "post",
   initialState,
   reducers: {
-    setPosts: (_, action: PayloadAction<Post[]>) => {
-      return action.payload;
-    },
-    updatePost: (state, action: PayloadAction<Post>) => {
-      const newPosts = state.map((o) => {
-        if (o.postId === action.payload.postId) {
-          return action.payload;
+    addReply: (state, action: PayloadAction<Post>) => {
+      const newReply = {
+        ...action.payload,
+        numberOfLikes: 0,
+        numberOfReplies: 0,
+      };
+      const newPosts = state.posts.map((o) => {
+        if (o.postId === action.payload.parentPostId) {
+          return {
+            ...o,
+            numberOfReplies: o.numberOfReplies + 1,
+          };
         }
         return o;
       });
-      return newPosts;
+      state.posts = [newReply, ...newPosts];
+      if (action.payload.parentPostId === state.expandedPost.postId) {
+        state.expandedPost.numberOfReplies++;
+      }
     },
     appendPost: (state, action: PayloadAction<Post>) => {
-      state.unshift({ ...action.payload, numberOfLikes: 0 });
-      return state;
+      state.posts.unshift({
+        ...action.payload,
+        numberOfLikes: 0,
+        numberOfReplies: 0,
+      });
+    },
+    setPosts: (state, action: PayloadAction<Post[]>) => {
+      state.posts = action.payload;
+    },
+    setExpandedPost: (state, action: PayloadAction<Post>) => {
+      state.expandedPost = action.payload;
+    },
+    toggleLikePost: (state, action: PayloadAction<number>) => {
+      const newPosts = state.posts.map((o) => {
+        if (o.postId === action.payload) {
+          const isLikedByCurrentUser = !o.isLikedByCurrentUser;
+          return {
+            ...o,
+            isLikedByCurrentUser,
+            numberOfLikes: isLikedByCurrentUser
+              ? o.numberOfLikes + 1
+              : o.numberOfLikes - 1,
+          };
+        }
+        return o;
+      });
+      state.posts = newPosts;
+
+      if (action.payload === state.expandedPost.postId) {
```

Comment:
nice, this removes the need for `updateExpandedPost` reducer I think

Reply:
Yup, `updateExpandedPost` was part of `expandedPostSlice.ts`, which got deleted